examples/src/main/kotlin/ForwardingService.kt

```diff
@@ -0,0 +1,220 @@
+        println("Transaction will be forwarded after it confirms.")
+        println("Waiting for confirmation...")
+        wallet.waitForConfirmations(incomingTx, REQUIRED_CONFIRMATIONS)
+            .thenCompose { confidence: TransactionConfidence ->
+                // Required confirmations received, now create and send forwarding transaction
+                System.out.printf("Incoming tx has received %d confirmations.\n", confidence.depthInBlocks)
+                forward(wallet, incomingTx, forwardingAddress)
+            }
+            .whenComplete { broadcast: TransactionBroadcast?, throwable: Throwable ->
+                if (broadcast != null) {
+                    System.out.printf(
+                        "Sent %s onwards and acknowledged by peers, via transaction %s\n",
+                        broadcast.transaction().outputSum.toFriendlyString(),
+                        broadcast.transaction().txId
+                    )
+                } else {
+                    println("Exception occurred: $throwable")
+                }
+            }
+    }
+
+    /**
+     * Forward an incoming transaction by creating a new transaction, signing, and sending to the specified address.
+     * @param wallet The active wallet
+     * @param incomingTx the received transaction
+     * @param forwardingAddress the address to send to
+     * @return A future for a TransactionBroadcast object that completes when relay is acknowledged by peers
+     */
+    private fun forward(
+        wallet: Wallet,
+        incomingTx: Transaction,
+        forwardingAddress: Address?
+    ): CompletableFuture<TransactionBroadcast?> {
+        // Send coins received in incomingTx onwards by sending exactly the outputs that have been sent to us
+        val sendRequest = SendRequest.emptyWallet(forwardingAddress)
+        sendRequest.coinSelector = forwardingCoinSelector(incomingTx.txId)
+        System.out.printf("Creating outgoing transaction for %s...\n", forwardingAddress)
+        return wallet.sendTransaction(sendRequest)
+            .thenCompose { broadcast: TransactionBroadcast ->
+                System.out.printf(
+                    "Transaction %s is signed and is being delivered to %s...\n",
+                    broadcast.transaction().txId,
+                    network
+                )
+                broadcast.awaitRelayed() // Wait until peers report they have seen the transaction
+            }
+    }
+
+    companion object {
+        const val USAGE = "Usage: address-to-forward-to [mainnet|testnet|signet|regtest]"
+        const val REQUIRED_CONFIRMATIONS = 1
+        const val MAX_CONNECTIONS = 4
+
+        /**
+         * Run the forwarding service as a command line tool
+         * @param args See [.USAGE]
+         */
+        @JvmStatic
+        fun main(args: Array<String>) { //pass the network and forwarding address in this order [address, network]
```

Comment:
I wonder if we can make `main` a top-level function in Kotlin. Similar to how it can be done in Java 22 EA.